contrib/platform/src/com/sun/jna/platform/win32/Advapi32.java

```diff
@@ -284,6 +301,292 @@ boolean CreateWellKnownSid(int wellKnownSidType, PSID domainSid,
                                PSID pSid, IntByReference cbSid);
 
     /**
+     * The InitializeSecurityDescriptor function initializes a new security descriptor.
+     * @param pSecurityDescriptor
+     *              A pointer to a SECURITY_DESCRIPTOR structure that the function initializes.
+     * @param dwRevision
+     *              The revision level to assign to the security descriptor. This parameter
+     *              must be SECURITY_DESCRIPTOR_REVISION.
+     * @return If the function succeeds, the return value is nonzero. If the
+     *         function fails, the return value is zero. For extended error
+     *         information, call GetLastError.
+     */
+    boolean InitializeSecurityDescriptor(SECURITY_DESCRIPTOR pSecurityDescriptor, int dwRevision);
+
+    /**
+     * The GetSecurityDescriptorControl function retrieves a security descriptor control and revision information.
+     * @param pSecurityDescriptor
+     *              A pointer to a SECURITY_DESCRIPTOR structure whose control and revision
+     *              information the function retrieves.
+     * @param pControl
+     *              A pointer to a SECURITY_DESCRIPTOR_CONTROL structure that receives the security descriptor's
+     *              control information.
+     * @param lpdwRevision
+     *              A pointer to a variable that receives the security descriptor's revision value.
+     *              This value is always set, even when GetSecurityDescriptorControl returns an error.
+     * @return If the function succeeds, the return value is nonzero. If the
+     *         function fails, the return value is zero. For extended error
+     *         information, call GetLastError.
+     */
+    boolean GetSecurityDescriptorControl(SECURITY_DESCRIPTOR pSecurityDescriptor, IntByReference pControl, IntByReference lpdwRevision);
```

Comment:
pControl is a PSECURITY_DESCRIPTOR_CONTROL is defined as:

```h
typedef WORD SECURITY_DESCRIPTOR_CONTROL, *PSECURITY_DESCRIPTOR_CONTROL;
```

Shouldn't this be ShortByReference?